### PR TITLE
Fix negative values in charts

### PR DESCRIPTION
**Pull request by Njektt**

### CODE REVIEW

**Comment by evshvarov** [comment]
Which issue it solves?

**Comment by Njektt** [comment]
> Which issue it solves?

https://github.com/intersystems-community/DeepSeeWeb/issues/127

**Comment by Njektt** [comment]
Before
image

After
image

**Comment by gnibeda** [comment]
We can't remove the axis minimum value due to the requirement that the axis should start from 0. Removing it and having a minimum data value of 0.3 will produce an axis with a 0.3 minimum value, but not zero.

I suggest the following:
1. Check all data values if there are no negative values
2. If yes - set the minimum axis value to zero

example:
```typescript
if (!data.some(d => d < 0)) {
    yAxis.min = 0;
}
```

File: src/app/components/widgets/charts/base-chart.class.ts [hunk 1]
```diff
@@ -1299,7 +1299,6 @@ export class BaseChartClass extends BaseWidget implements OnInit, AfterViewInit
         const yAxis = this.chartConfig.yAxis as Highcharts.YAxisOptions;
         const xAxis = this.chartConfig.xAxis as Highcharts.XAxisOptions;
 
-        yAxis.min = 0;
```
Comment (gnibeda) [inline src/app/components/widgets/charts/base-chart.class.ts:1302]: We need to check negative values first. More details in the comment

**Review by gnibeda — CHANGES_REQUESTED** [review]

File: src/app/components/widgets/charts/base-chart.class.ts (hunk 1 above)
Comment (Njektt) [reply src/app/components/widgets/charts/base-chart.class.ts:1302]: Done

**Comment by evshvarov** [comment]
Great stuff!
Maybe it makes sense to add it into CONTRIBUTION.MD?

**Comment by gnibeda** [comment]
`changelog.md` is used by UI to show the "what's new" page. no need to add into "CONTRIBUTION.MD"

File: src/changelog.md [hunk 2]
```diff
@@ -1,3 +1,6 @@
+#### 3.1.46
```
Comment (gnibeda) [inline src/changelog.md:1]: Wrong version, should be 3.1.47. No changes needed, I'll fix it

**Review by gnibeda — APPROVED** [review]

**Pull request merged**